File: Singleton/uml-generated-code/Singleton.java


/**
 * Class Singleton
 */
public class Singleton {

  //
  // Fields
  //

  static private Singleton instance;
  private int data;
  
  //
  // Other methods
  //

  /**
   * @return       Singleton
   */
  public static synchronized Singleton getInstance()
  {
      if(instance == null)
          instance = new Singleton();

      return instance;
  }

  public void display() {
      System.out.println(Integer.toString(data));
  }

  public void setData(int data) {
      this.data = data;
  }

  /**
   */
  private void Singleton()
  {
  }


}

File: Singleton/uml-generated-code/Singleton.cpp
#include "Singleton.h"

// Constructors/Destructors
//  

Singleton::Singleton () {
    data = 10;
//initAttributes();
}

Singleton::~Singleton () { }

//  
// Methods
//  
void Singleton::display() {
    std::cout << data << std::endl;
}

std::weak_ptr<Singleton> Singleton::__instance = std::weak_ptr<Singleton>();

// Accessor methods
//  

// Other methods
//  

//void Singleton::initAttributes () {
//}

